lib/pbench/test/unit/agent/task/test_add_metalog_option.py

```diff
@@ -0,0 +1,43 @@
+import configparser
+from pathlib import Path
+
+from pbench.cli.agent.commands.log import add_metalog_option
+
+
+class TestAddMetalogOption:
+    @staticmethod
+    def test_add_metalog_option(pytestconfig):
+        TMP = pytestconfig.cache.get("TMP", None)
+        mdlog = Path(TMP) / "metadata.log"
+        mdlog.write_text(
+            """[existing]
+            existing_option = 41
+            """
+        )
+        # New section, new option
+        add_metalog_option(mdlog, "new", "new_option", "420")
+        # Existing section, new option
+        add_metalog_option(mdlog, "existing", "new_option", "142")
+        # Existing section, existing option
+        add_metalog_option(mdlog, "existing", "existing_option", "42")
+
+        cfg = configparser.ConfigParser()
+        cfg.read(mdlog)
+
+        sections = cfg.sections()
+        assert len(sections) == 2
+        assert "new" in sections
+        assert "existing" in sections
+
+        options = cfg.options("new")
+        assert len(options) == 1
+        assert "new_option" in options
+
+        options = cfg.options("existing")
+        assert len(options) == 2
+        assert "new_option" in options
+        assert "existing_option" in options
+
+        assert cfg.get("new", "new_option") == "420"
+        assert cfg.get("existing", "new_option") == "142"
+        assert cfg.get("existing", "existing_option") == "42"
```

Comment:
While it's fine to test the final state, it seems like it would also be good to test the state between each addition.